### PR TITLE
DB: FR3 validation rule and testcase

### PR DESCRIPTION
**Pull request by allenbross-wf**
#### Reason for change

> DBA.FR3: The company's accounting class is missing. This must be marked separately in the field (fsa:ClassOfReportingEntity).

#### Description of change

Added a validation rule to check for the existence of a fact with a value. 

#### Steps to Test

CI

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: arelle/plugin/validate/DBA/rules/__init__.py [hunk 1]
```diff
@@ -52,6 +52,27 @@ def errorOnDateFactComparison(
...
+        message: str,
+) -> Iterable[Validation]:
+    """
+    Yields an error if a fact with the given QName is not tagged with a valid date and a non-nil value.
+    :return: Yields validation errors.
+    """
+    facts: set[ModelFact] = modelXbrl.factsByQname.get(factQn, set())
+    if facts:
+        for fact in facts:
+            if fact.xValue is not None:
```
Comment (derekgengenbacher-wf) [inline arelle/plugin/validate/DBA/rules/__init__.py:68]: In the description above it states that a fact must be tagged `with a valid date and a non-nil value.` But you dont check that the facts date is acceptable in this code. Is that done elsewhere?

Comment (allenbross-wf) [reply arelle/plugin/validate/DBA/rules/__init__.py:68]: That is a relic of a mis-reading of the spec.

**Review by austinmatherne-wk — DISMISSED** [review]

File: arelle/plugin/validate/DBA/rules/__init__.py [hunk 2]
```diff
@@ -52,6 +52,27 @@ def errorOnDateFactComparison(
...
+        message: str,
+) -> Iterable[Validation]:
+    """
+    Yields an error if a fact with the given QName is not tagged with a non-nil value.
+    :return: Yields validation errors.
+    """
+    facts: set[ModelFact] = modelXbrl.factsByQname.get(factQn, set())
+    if facts:
+        for fact in facts:
+            if fact.xValue is not None:
```
Comment (aaroncameron-wk) [inline arelle/plugin/validate/DBA/rules/__init__.py:68]: I _think_ checking for the `xsi:nil` attribute (ModelFact.isNil [here](https://github.com/Arelle/Arelle/blob/672eed35cb54b8142f2d6facdd2127ff056c0592/arelle/ModelInstanceObject.py#L353)) is what we should interpret "non-nil" to mean in the rule text. That's at least what we've done [elsewhere](https://github.com/Arelle/Arelle/blob/672eed35cb54b8142f2d6facdd2127ff056c0592/arelle/plugin/validate/DBA/rules/fr.py#L224).

File: tests/resources/conformance_suites/dba/fr/fr3-testcase.xml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xmlns="http://xbrl.org/2005/conformance"
+        name="DBA.FR3"
+        description="DBA.FR3: The company's accounting class is missing. This must be marked separately in the field (fsa:ClassOfReportingEntity)"
+        outpath=''
+        owner="support@arelle.org"
+        xsi:schemaLocation="http://xbrl.org/2005/conformance https://www.xbrl.org/2005/conformance.xsd">
+    <variation id="invalid" name="Invalid">
+        <description>
+            The end date of the accounting period is before the start date of the accounting period.
```
Comment (aaroncameron-wk) [inline tests/resources/conformance_suites/dba/fr/fr3-testcase.xml:13]: Left over from copy-paste

**Review by aaroncameron-wk — APPROVED** [review]

**Review by austinmatherne-wk — APPROVED** [review]

**Pull request merged**